torch/onnx/errors.py

```diff
@@ -54,15 +59,38 @@ def __init__(
                     f"Support for this operator was added in version {supported_version}, "
                     "try exporting with this version."
                 )
+                diagnostics.context.diagnose(
+                    diagnostics.rules.operator_supported_in_newer_opset_version,
+                    diagnostics.levels.ERROR,
+                    message_args=(
+                        f"{domain}::{op_name}",
+                        version,
+                        supported_version,
+                    ),
+                )
             else:
                 msg += "Please feel free to request support or submit a pull request on PyTorch GitHub: "
                 msg += _constants.PYTORCH_GITHUB_ISSUES_URL
+                diagnostics.context.diagnose(
+                    diagnostics.rules.missing_standard_symbolic_function,
+                    diagnostics.levels.ERROR,
+                    message_args=(
+                        f"{domain}::{op_name}",
+                        version,
+                        _constants.PYTORCH_GITHUB_ISSUES_URL,
+                    ),
+                )
         else:
             msg = (
                 f"ONNX export failed on an operator with unrecognized namespace '{domain}::{op_name}'. "
                 "If you are trying to export a custom operator, make sure you registered "
                 "it with the right domain and version."
             )
+            diagnostics.context.diagnose(
+                diagnostics.rules.missing_custom_symbolic_function,
+                diagnostics.levels.ERROR,
+                message_args=(f"{domain}::{op_name}",),
+            )
```

Comment:
Would it be possible to get the message (for the python error) from the rule in the future?

Reply:
Tricky to have a good ux experience with message and arguments. Getting message alone is simple, but it's one step(f12) away from knowing what exact arguments it's expecting. 
Tried a few approaches locally. The best ux friendly one is generating individual class for each rule with "overloaded" arguments. LMK if you have ideas...

Reply:
Thanks for the explanation! I will think about it